podman/domain/containers_run.py

```diff
@@ -77,7 +77,7 @@ def run(
         if log_type in ("json-file", "journald"):
             log_iter = container.logs(stdout=stdout, stderr=stderr, stream=True, follow=True)
 
-        exit_status = container.wait()["StatusCode"]
+        exit_status = container.wait()
         if exit_status != 0:
```

Comment:
how does this check still work if you remove the ["StatusCode"] aspect? now you are checking `if DICT != 0`

Reply:
def wait(self, **kwargs) -> Dict[Literal["StatusCode", "Error"], Any]:
Looks like to original code is correct

Reply:
Wait wait wait, there is definitely an issue here.
By default, `wait()` in podman-py uses the libpod operation  [ContainerwaitLibPod](https://docs.podman.io/en/latest/_static/api.html#tag/containers/operation/ContainerWaitLibpod), which returns a JSON-encoded integer. For that case, the type annotation
```
def wait(self, **kwargs) -> Dict[Literal["StatusCode", "Error"], Any]:
```
isn't correct. That annotation matches the podman compat API, where [ContainerWait](https://docs.podman.io/en/latest/_static/api.html#tag/containers-(compat)/operation/ContainerWait) returns a JSON object with the expected Keys `Error` and `StatusCode`.
I just discovered podman-py and don't know the code base enough yet to fix it myself, but would recommend to look into that issue again. I got here having experienced the same issue as in #201

Reply:
@cdoern PTAL